kratos/includes/properties.h

```diff
@@ -277,30 +295,138 @@ class Properties : public IndexedObject
         return result_key;
     }
 
+    /**
+     * @brief This method returns the number of subproperties
+     * @return The current number of subproperties
+     */
+    std::size_t NumberOfSubproperties()
+    {
+        return mSubPropertiesList.size();
+    }
+
+    /**
+     * @brief This method insert a new property into the list of subproperties
+     * @param pNewSubProperty The new property to be added
+     */
+    void AddSubProperty(Properties::Pointer pNewSubProperty)
+    {
+        mSubPropertiesList.insert(mSubPropertiesList.begin(), pNewSubProperty);
+    }
+
+    /**
+     * @brief This method gets the subproperty from the index corresponding to the property id
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return The pointer to the subproperty of interest
+     */
+    Properties::Pointer pGetSubProperty(IndexType SubPropertyIndex)
+    {
+        return mSubPropertiesList(SubPropertyIndex);
+    }
+
+    /**
+     * @brief This method gets the subproperty from the index corresponding to the property id
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return The reference to the subproperty of interest
+     */
+    Properties& GetSubProperty(IndexType SubPropertyIndex)
+    {
+        return *(mSubPropertiesList(SubPropertyIndex));
+    }
+
+    /**
+     * @brief This method gets the subproperty from the index corresponding to the property id
+     * @param SubPropertyIndex The index of the subproperty to be get
+     * @return The reference to the subproperty of interest
+     */
+    Properties& GetSubProperty(IndexType SubPropertyIndex) const
+    {
+        return *(mSubPropertiesList.find(SubPropertyIndex));
+    }
+
+    /**
+     * @brief This method returns the whole list of subproperties
+     * @return The whole lis of subproperties
+     */
+    PropertiesContainerType& GetSubProperties()
```

Comment:
so this gives a pointer to the list?

Reply:
It gives a PointerVectorSet

https://github.com/KratosMultiphysics/Kratos/blob/80332ee8dec355c6d78e9d153df521c68f90d000/kratos/includes/properties.h#L95

Reply:
You told me to do exactly that